packages/file-asset-apis/src/s3/createS3Client.ts

```diff
@@ -0,0 +1,60 @@
+import fs from 'fs-extra';
+import { Agent } from 'https';
+import { S3Client as BaseClient } from '@aws-sdk/client-s3';
+import { NodeHttpHandler } from '@aws-sdk/node-http-handler';
+import type { S3ClientConfig as baseConfig } from '@aws-sdk/client-s3';
+import { debugLogger, has } from '@terascope/utils';
+import type { S3Client } from './client-types';
+
+export interface S3ClientConfig extends baseConfig {
+    sslEnabled?: boolean,
+    certLocation?: string,
+    httpOptions?: object,
+    secretAccessKey?: string,
+    accessKeyId?: string
+}
+
+export async function createS3Client(
+    config: S3ClientConfig,
+    logger = debugLogger('elasticsearch-client')
+): Promise<S3Client> {
+    logger.info(`Using S3 endpoint: ${config.endpoint}`);
+    // pull certLocation from env
+    // https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
+    // Instead of updating the client, we can just update the config before creating the client
+    if (config.sslEnabled) {
+        const certPath = config.certLocation ?? '/etc/ssl/certs/ca-certificates.crt';
+        const pathFound = await fs.exists(certPath);
+
+        if (!pathFound) {
+            throw new Error(
+                `No cert path was found in config.certLocation: "${config.certLocation}" or in default "/etc/ssl/certs/ca-certificates.crt" location`
+            );
+        }
+        // Assumes all certs needed are in a single bundle
+        const certs = await fs.readFile(certPath);
+
+        const configHttpOptions = config.httpOptions ?? {};
+
+        const httpOptions = Object.assign(
+            { rejectUnauthorized: true },
```

Comment:
In general we would want this setting to be `true`, but there might be scenarios where we want this configurable on the connector.  For instance, in a test scenario I am working on right now, it would be nice to at least test and see if it works with `rejectUnauthorized: false`.  If you can easily make this configurable, with a default to `true` that would be cool, or we can leave it for future work.

Reply:
You can do that by passing in `httpOptions: { rejectUnauthorized: false }` in your config.